internal/dag/dag.go

```diff
@@ -303,6 +303,10 @@ func (s *Secret) PrivateKey() []byte {
 	return s.Object.Data[v1.TLSPrivateKeyKey]
 }
 
+func (s *Secret) isValid() bool {
+	return s.Cert() != nil && len(s.Cert()) > 0 && s.PrivateKey() != nil && len(s.PrivateKey()) > 0
```

Comment:
Thank you but I don't think this is right.

We should not be creating an invalid secret object and placing it in the dag. Either we filter this out here

https://github.com/heptio/contour/blob/master/internal/dag/builder.go#L570

which is not my preference because it encodes knowledge that secret objects are only present at some levels.

Or we filter it out earlier in `lookupSecret`. I'd like to try that. Can you please make this a function

```go
// validSecret returns true if the Secret contains 
func validSecret(s *v1.Secret) bool {
    return len(s.Data[v1.TLSCertKey]) > 0 && len(s.Data[v1.TLSPrivateKeyKey]) > 0
}
```
Also, len(s.Cert()) will work fine without the nil check.

Reply:
Sure. Thanks for the feedback!